### PR TITLE
ci(chglog): Update chglog and clean up config

### PR DESCRIPTION
**Pull request by andytom**
#### Description

Have updated the version of chglog we use in the deployment pipeline to
the most recent version and updated the config order the config
correctly.

#### Motivation and Context

Related to #3205

#### How Has This Been Tested?

- [ ] I have tested using **MacOS**
- [x] I have tested using **Linux**
- [ ] I have tested using **Windows**

#### Checklist:

- [ ] I have updated the documentation accordingly.
- [ ] I have updated the tests accordingly.

### CODE REVIEW

File: .github/workflows/deploy.yml [hunk 1]
```diff
@@ -136,7 +136,7 @@ jobs:
 
       - name: Setup | Release notes
         run: |
-          GO111MODULE=on go get github.com/git-chglog/git-chglog/cmd/git-chglog@0.9.1
+          GO111MODULE=on go get github.com/git-chglog/git-chglog/cmd/git-chglog@0.15.0
```
Comment (davidkna) [inline .github/workflows/deploy.yml:139]: If you're updating the chglog version, you might also need to update the go version.

Comment (davidkna) [reply .github/workflows/deploy.yml:139]: In addition, the tag needs to be changed and it looks like `go get` has been deprecated in favour of `go install`:
```suggestion
          go install github.com/git-chglog/git-chglog/cmd/git-chglog@v0.15.0
```

Comment (andytom) [reply .github/workflows/deploy.yml]: I've bumped the version of go to the same version that `git-chglog` use in their [CI workflow](https://github.com/git-chglog/git-chglog/blob/master/.github/workflows/ci.yml#L15).

Thanks I didn't know `go get` was deprecated, I've been out of the Go ecosystem for longer than I thought.

**Review by davidkna — APPROVED** [review]

File: .github/workflows/deploy.yml (hunk 1 above)
Comment (davidkna) [reply .github/workflows/deploy.yml]: I think the warning was added with go 1.17, so this seems to a fairly recent development.

**Pull request merged**